rosbag2_storage_sqlite3/src/rosbag2_storage_sqlite3/sqlite_storage.cpp

```diff
@@ -681,10 +703,16 @@ void SqliteStorage::read_metadata()
         rcutils_time_point_value_t, std::string, std::string>();
 
       for (auto result : query_results) {
+        auto yaml_node = YAML::Load(std::get<6>(result));
```

Comment:
Here, a conversion from serialized YAML (`std::string`) & `std::vector<rclcpp::QoS>` is needed.